src/software/firmware/srcs/activation.cpp

```diff
@@ -47,3 +47,13 @@ void ActivationController::onStopButton() {
         break;
     }
 }
+
+void ActivationController::refreshState() {
+    // If the 2nd STOP deadline is exceeded, switch back
+    // to running state. This is needed to make sure millis() counter
+    // overflow does not make the test invalid.
+    if ((m_state == RUNNING_READY_TO_STOP)
+        && ((millis() - m_timeOfLastStopPushed) >= SECOND_STOP_MAX_DELAY_MS)) {
+        m_state = RUNNING;
+    }
+}
```

Comment:
We need an empty line at the end of the file.
If you don't have time, we can take care of this point. 

PS: Thanks for all your PRs and reviews.

Reply:
I am on it. Still waiting for my push to finish

Reply:
I dont know why I cannot push/fetch my remote. I let you make the change then.